src/NovelRT/Graphics/BasicFillRect.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#include <NovelRT.h>
+
+namespace NovelRT::Graphics {
+
+  BasicFillRect::BasicFillRect(const Transform& transform,
+    int layer,
+    Camera* camera,
+    ShaderProgram shaderProgram,
+    const RGBAConfig& fillColour) :
+    RenderObject(transform, layer, shaderProgram, camera), _colourConfig(fillColour),
+    _colourBuffer(Utilities::Lazy<GLuint>(generateStandardBuffer)) {}
+
+  void BasicFillRect::drawObject() {
```

Comment:
just a note: a lot of this is consistent between the different render objects and could be abstracted
e.g. the real differentiation is the shader programs being set and the number of vertices/indices

Reply:
Considering our approach to restructure buffer usage, I think this can wait until then. I'd rather not break things any further.